src/pages/index.tsx

```diff
@@ -6,17 +6,17 @@ import { useAppSelector } from '@/store'
 import useChainId from '@/hooks/useChainId'
 import chains from '@/config/chains'
 
-const useLastSafe = (): string | undefined => {
-  const chainId = useChainId()
+const useLastSafe = (chainId: string): string | undefined => {
   const lastSafeAddress = useAppSelector((state) => selectLastSafeAddress(state, chainId))
   const prefix = Object.keys(chains).find((prefix) => chains[prefix] === chainId)
   return prefix && lastSafeAddress ? `${prefix}:${lastSafeAddress}` : undefined
 }
 
 const IndexPage: NextPage = () => {
   const router = useRouter()
+  const chainId = useChainId()
   const { chain } = router.query
-  const lastSafe = useLastSafe()
+  const lastSafe = useLastSafe(typeof chain === 'string' ? chains[chain] : chainId)
```

Comment:
I believe `useChainId` already checks for `chains[chain]` with the query param so we could do `useLastSafe(chainId)` as well.